src/Server/modules/USPlacesManager.js

```diff
@@ -0,0 +1,264 @@
+					var temp = new Date(key);
+					var d =
+						temp.getMonth() +
+						1 +
+						'/' +
+						temp.getDate() +
+						'/' +
+						temp.getFullYear();
+					if (d == date) {
+						delete allUSPlaces[i].numDeaths[key];
+						result = true;
+						break;
+					}
+				}
+			}
+		}
+	}
+	return result;
+}
+// function SaveRecords()
+// A void function that saves allUSPlace (Maybe modified) into csv.
+function SaveRecords(){
+    writeCSVModule.RecordUSData(allUSPlaces);
+}
+
+function GetTwoPlacesComparison(state1, county1, state2, county2, typeOfData) {
+    var county1Data = null;
+    var county2Data = null;
+
+        for (var i = 0; i < allUSPlaces.length; ++i) {
+            if (allUSPlaces[i].state == state1 && allUSPlaces[i].county == county1) {
+                county1Data = allUSPlaces[i];
+            }
+            if (allUSPlaces[i].state == state2 && allUSPlaces[i].county == county2) {
+                county2Data = allUSPlaces[i];
+            }
+        }
+
+    if (county1Data == null || county2Data == null) {
+        place1Value = (county1Data == null) ? -1 : 0;
+        place2Value = (county2Data == null) ? -1 : 0;
+        return { place1Value: place1Value, place2Value: place2Value }
+    }
+
+    var place1Value, place2Value;
+
+    switch (typeOfData) {
+        case '1': // confirmed
+            place1Value = county1Data.currentNumConfirmed;
+            place2Value = county2Data.currentNumConfirmed;
+            break;
+        case '2': // dead
+            place1Value = county1Data.currentNumDeaths;
+            place2Value = county2Data.currentNumDeaths;
+            break;
+        case '3': // recovered
+            place1Value = county1Data.currentNumRecovered;
+            place2Value = county2Data.currentNumRecovered;
+            break;
+    }
+
+    return { place1Value: place1Value, place2Value: place2Value }
+}
+
+
+function GetUSPopulationAnalysis(state, county) {
+    var unaffected, deaths, confirmed;
+
+    var countyData = null;
+
+    for (var i = 0; i < allUSPlaces.length; ++i) {
+        if (allUSPlaces[i].state == state && allUSPlaces[i].county == county) {
+            countyData = allUSPlaces[i];
+        }
+    }
+
+    if (countyData == null) { // The requested county does not exist.
+        return { unaffected: -1, deaths: -1, confirmed: -1  }
+    }
+
+    deaths = countyData.currentNumDeaths;
+    confirmed = countyData.currentNumConfirmed;
+    unaffected = countyData.population - deaths - confirmed;
+
+    return { unaffected: unaffected, deaths: deaths, confirmed: confirmed }
+}
+
+// function GetRows(county, state, typeOfData)
+// Input: county (string), state (string), typeOfData (int) (1 = conf, 2 = deaths)
+// Similar to what we do in routes/USData.js.
+// We want to return “row”.
+// We want to use this function in routes/USData.js.
+function GetRows(county, state, typeOfData){
+    //How to combine different requests into one function?
```

Comment:
We want to do the same thing as in Server/routes/USData.js line 16 - 64.

In Server/routes/USData.js line 16 - 64, we get a request from the client-side (req.query.Country, req.query.State, and req.query.TypeOfData) and sends "row".

Instead, here, we want to get country, state, and typeOfData from the argument and returns "row".

Please let me know if still unclear :D